spring-boot/src/test/java/org/springframework/boot/ImageBannerTests.java

```diff
@@ -0,0 +1,239 @@
+
+		assertThat("Banner contains BLACK", banner,
+				containsString(getAnsiOutput(AnsiColor.BLACK)));
+		assertThat("Banner contains RED", banner,
+				containsString(getAnsiOutput(AnsiColor.RED)));
+		assertThat("Banner contains GREEN", banner,
+				containsString(getAnsiOutput(AnsiColor.GREEN)));
+		assertThat("Banner contains YELLOW", banner,
+				containsString(getAnsiOutput(AnsiColor.YELLOW)));
+		assertThat("Banner contains BLUE", banner,
+				containsString(getAnsiOutput(AnsiColor.BLUE)));
+		assertThat("Banner contains MAGENTA", banner,
+				containsString(getAnsiOutput(AnsiColor.MAGENTA)));
+		assertThat("Banner contains CYAN", banner,
+				containsString(getAnsiOutput(AnsiColor.CYAN)));
+		assertThat("Banner contains WHITE", banner,
+				containsString(getAnsiOutput(AnsiColor.WHITE)));
+
+		assertThat("Banner contains BRIGHT_BLACK", banner,
+				containsString(getAnsiOutput(AnsiColor.BRIGHT_BLACK)));
+		assertThat("Banner contains BRIGHT_RED", banner,
+				containsString(getAnsiOutput(AnsiColor.BRIGHT_RED)));
+		assertThat("Banner contains BRIGHT_GREEN", banner,
+				containsString(getAnsiOutput(AnsiColor.BRIGHT_GREEN)));
+		assertThat("Banner contains BRIGHT_YELLOW", banner,
+				containsString(getAnsiOutput(AnsiColor.BRIGHT_YELLOW)));
+		assertThat("Banner contains BRIGHT_BLUE", banner,
+				containsString(getAnsiOutput(AnsiColor.BRIGHT_BLUE)));
+		assertThat("Banner contains BRIGHT_MAGENTA", banner,
+				containsString(getAnsiOutput(AnsiColor.BRIGHT_MAGENTA)));
+		assertThat("Banner contains BRIGHT_CYAN", banner,
+				containsString(getAnsiOutput(AnsiColor.BRIGHT_CYAN)));
+		assertThat("Banner contains BRIGHT_WHITE", banner,
+				containsString(getAnsiOutput(AnsiColor.BRIGHT_WHITE)));
+	}
+
+	@Test
+	public void renderSimpleGradient() {
+		AnsiOutput.setEnabled(AnsiOutput.Enabled.NEVER);
+		String banner = printBanner(IMAGE_BANNER_GRADIENT);
+		String expectedResult = "@#8&o:*. ";
+
+		assertThat(banner, startsWith(expectedResult));
+	}
+
+	@Test
+	public void renderBannerWithDefaultAspectRatio() {
+		String banner = printBanner(IMAGE_BANNER_BLACK_AND_WHITE);
+		int bannerHeight = getBannerHeight(banner);
+
+		assertThat(bannerHeight, equalTo(2));
+	}
+
+	@Test
+	public void renderBannerWithCustomAspectRatio() {
+		setAspectRatio(1.0d);
+		String banner = printBanner(IMAGE_BANNER_BLACK_AND_WHITE);
+		int bannerHeight = getBannerHeight(banner);
+
+		assertThat(bannerHeight, equalTo(4));
+	}
+
+	@Test
+	public void renderLargeBanner() {
+		String banner = printBanner(IMAGE_BANNER_LARGE);
+		int bannerWidth = getBannerWidth(banner);
+
+		assertThat(bannerWidth, equalTo(72));
+	}
+
+	@Test
+	public void renderLargeBannerWithACustomWidth() {
+		setMaxWidth(60);
+		String banner = printBanner(IMAGE_BANNER_LARGE);
+		int bannerWidth = getBannerWidth(banner);
+
+		assertThat(bannerWidth, equalTo(60));
+	}
+
+	private int getBannerHeight(String banner) {
+		return banner.split("\n").length;
```

Comment:
Could you `.trim()` this first?
For me the tests fail because it's one line too long.
(Windows):
```
"[49m[30m@[39m
[49m[30m@[39m
[49m[97m [39m
[49m[97m [39m

"
```